src/store/index.js

```diff
@@ -112,13 +112,23 @@ export default new Vuex.Store({
             state.gameState.playerInfo = !state.gameState.playerInfo
         },
         stepTwoToQuestPhase: ({state}) => {
-            state.gameState.playerInfo = !state.gameState.playerInfo
-            state.gameState.questInfoDisplay = !state.gameState.questInfoDisplay
-            state.gameState.proposeMissionParty = !state.gameState.proposeMissionParty
+            //This means that we're already past the ProposeMissionParty state so we jump straight to ProposedPartyVote
+            if (state.ProposedPartyVoteMenu.proposedParty.length >= 1) {
+                state.gameState.playerInfo = false
+                state.gameState.questInfoDisplay = true
+                state.gameState.proposedPartyVote = true
+            } else {
+                state.gameState.playerInfo = false
+                state.gameState.questInfoDisplay = true
+                state.gameState.proposeMissionParty = true
+            }
         },
         ToggleProposeMissionPartyAndProposedPartyVote: ({state}) => {
-            state.gameState.proposeMissionParty = !state.gameState.proposeMissionParty
-            state.gameState.proposedPartyVote = !state.gameState.proposedPartyVote
+            //If we are in PlayerInfo state we cannot move forward until the user has Ready'd up
+            if (!state.gameState.playerInfo) {
```

Comment:
I would handle this check inside the Avalon component, feels more natural since that's where most of the logic that effects state lives.